skymp5-server/cpp/server_guest_lib/ActionListener.cpp

```diff
@@ -449,3 +449,19 @@ void ActionListener::OnCustomEvent(const RawMessageData& rawMsgData,
     listener->OnMpApiEvent(eventName, args, ac->GetFormId());
   }
 }
+
+void ActionListener::OnChangeValues(const RawMessageData& rawMsgData,
+                                    const float healthPercentage,
+                                    const float magickaPercentage,
+                                    const float staminaPercentage)
+{
+  MpActor* actor = partOne.serverState.ActorByUser(rawMsgData.userId);
+  if (!actor) {
+    throw std::runtime_error("Unable to change values without Actor attached");
+  }
+  MpChangeForm changeForm = actor->GetChangeForm();
+  changeForm.healthPercentage = healthPercentage;
+  changeForm.magickaPercentage = magickaPercentage;
+  changeForm.staminaPercentage = staminaPercentage;
+  actor->ApplyChangeForm(changeForm);
```

Comment:
It seems that it works, but `ApplyChangeForm` is too heavy to use there. It's used to apply all character data loaded from a database. As far as I remember it also doesn't save change form automatically. Consider adding a method to the `MpActor` class that would update actor value percentages. Example: https://github.com/skyrim-multiplayer/skymp/blob/51b3da52de3d1d2ba5ddc18895112538f48a2264/skymp5-server/cpp/server_guest_lib/MpActor.cpp#L28